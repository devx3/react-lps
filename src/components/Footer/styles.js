import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.div`
    text-align: center;
    ${({ theme }) => css`
        border-top: 1px solid ${theme.colors.mediumGrey};
        a {
            text-decoration: none;
        }

        & ${TextComponent} p {
            font-size: ${theme.fonts.sizes.xsmall};
        }

        & ${SectionContainer} {
            padding-top: 0;
            padding-bottom: 0;
        }
    `};
`;
