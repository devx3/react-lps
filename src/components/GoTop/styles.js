import styled, { css } from 'styled-components';

export const Container = styled.a`
    ${({ theme }) => css`
        background: ${theme.colors.secondaryColor};
        color: ${theme.colors.primaryColor};
        display: flex;
        align-items: center;
        justify-content: center;
        position: fixed;
        width: 4rem;
        height: 4rem;
        bottom: 2rem;
        right: 2rem;
        z-index: 6;
        cursor: pointer;
    `}
`;
