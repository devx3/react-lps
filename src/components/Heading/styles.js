import styled, { css } from 'styled-components';

const titleSize = (theme, size) => {
    return css`
        font-size: ${eval(`theme.fonts.sizes.${size}`)};
        ${mediaFont(theme)}
    `;
};

const mediaFont = (theme) => css`
    @media ${theme.media.ltMedia} {
        font-size: ${theme.fonts.sizes.biggest};
    }
`;

const titleCase = (uppercase) => css`
    text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
    ${({ theme, darkMode, size, uppercase }) => css`
        color: ${darkMode ? theme.colors.white : theme.colors.primaryColor};
        ${titleSize(theme, size)};
        ${titleCase(uppercase)};
    `}
`;
