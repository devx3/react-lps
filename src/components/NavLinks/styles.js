import styled, { css } from 'styled-components';

export const Container = styled.nav`
    ${({ theme }) => css`
        display: flex;
        flex-flow: row wrap;

        @media ${theme.media.ltMedia} {
            flex-flow: column wrap;
            align-content: center;
        }
    `}
`;
