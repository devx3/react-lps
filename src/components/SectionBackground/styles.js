import styled, { css } from 'styled-components';

const backgroundActivateColor = (theme) => css`
    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
`;

export const Container = styled.div`
    ${({ theme, background }) => css`
        background: ${theme.colors.white};
        color: ${theme.colors.primaryColor};

        ${background && backgroundActivateColor(theme)}

        min-height: 100vh;
        display: flex;
        align-items: center;
    `}
`;
