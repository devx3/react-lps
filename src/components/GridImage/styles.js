import styled, { css } from 'styled-components';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
    ${({ theme }) => css`
        ${TextContainer} {
            margin-bottom: ${theme.spacings.xhuge};
        }
    `}
`;

export const Grid = styled.div`
    ${({ theme }) => css`
        counter-reset: grid-counter;
        display: grid;
        grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
        gap: ${theme.spacings.large};
    `}
`;

export const GridItem = styled.div`
    ${() => css`
        overflow: hidden;
    `}
`;

export const Image = styled.img`
    ${() => css`
        width: 100%;
        transition: all 300ms ease-in-out;

        &:hover {
            transform: scale(1.2) rotate(5deg);
        }
    `}
`;
