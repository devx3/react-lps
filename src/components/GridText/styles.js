import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
    ${({ theme }) => css`
        ${TextContainer} {
            margin-bottom: ${theme.spacings.xhuge};
        }
    `}
`;

export const Grid = styled.div`
    ${({ theme }) => css`
        counter-reset: grid-counter;
        display: grid;
        grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
        gap: ${theme.spacings.large};

        @media ${theme.media.ltMedia} {
            grid-template-columns: 1fr;
        }
    `}
`;

export const GridItem = styled.div`
    ${({ theme }) => css`
        ${HeadingContainer} {
            position: relative;
            left: 5rem;
            margin-bottom: 15px;
        }

        ${HeadingContainer}::before {
            counter-increment: grid-counter;
            content: counter(grid-counter);
            position: absolute;
            top: -2rem;
            font-size: ${theme.fonts.sizes.huge};
            left: -5rem;
            transform: rotate(-10deg);
        }
    `}
`;
