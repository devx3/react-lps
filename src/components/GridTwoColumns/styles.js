import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
    ${({ theme }) => css`
        display: grid;
        grid-template-columns: 1fr 2fr;
        align-items: center;
        gap: ${theme.spacings.large};

        @media ${theme.media.ltMedia} {
            grid-template-columns: 1fr;
        }
    `}
`;

export const TextContainer = styled.div`
    ${({ theme }) => css`
        @media ${theme.media.ltMedia} {
            margin-bottom: ${theme.spacings.large};
        }

        ${Heading} {
            margin-bottom: ${theme.spacings.large};
        }
    `}
`;

export const ImageContainer = styled.div`
    ${() => css``}
`;

export const Image = styled.img`
    ${() => css``}
`;
