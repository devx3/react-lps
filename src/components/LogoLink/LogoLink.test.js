import { render, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { LogoLink } from './index.jsx';

describe('<LogoLink />', () => {
    it('Should render as a text with a link', () => {
        renderTheme(<LogoLink text="Lucifer" link="http://localhost" />);
        const link = screen.getByText('Lucifer').parentNode;
        expect(link).toHaveAttribute('href', 'http://localhost');
    });
    it('Should render a image with a link', () => {
        renderTheme(
            <LogoLink
                text="Lucifer"
                srcImg="assets/images/logo.svg"
                link="http://localhost"
            />,
        );
        const link = screen.getByRole('img');
        expect(link).toHaveAttribute('src', 'assets/images/logo.svg');
        expect(link).toHaveAttribute('alt', 'Lucifer');
    });

    it('Should match snapshot text', () => {
        renderTheme(<LogoLink text="Lucifer" link="http://localhost" />);
        const link = screen.getByText('Lucifer').parentNode;
        expect(link).toMatchSnapshot();
    });

    it('Should match snapshot image', () => {
        renderTheme(
            <LogoLink
                text="Lucifer"
                srcImg="assets/images/logo.svg"
                link="http://localhost"
            />,
        );
        const link = screen.getByRole('img');
        expect(link).toMatchSnapshot();
    });
});
