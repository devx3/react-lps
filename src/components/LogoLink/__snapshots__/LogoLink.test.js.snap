// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLink /> Should match snapshot image 1`] = `
@media (max-width:768px) {

}

<img
  alt="Lucifer"
  src="assets/images/logo.svg"
/>
`;

exports[`<LogoLink /> Should match snapshot text 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c0 > img {
  height: 3rem;
}

@media (max-width:768px) {

}

<a
  class="c0"
  href="http://localhost"
>
  <span>
    Lucifer
  </span>
</a>
`;
